### PR TITLE
Introduced more nuanced point system for evaluating connections

### PR DESCRIPTION
**Pull request by dezok9**
## Description
- This pull request bolsters and builds upon the connections recommendation algorithm by introducing a point system for components that will go to produce a final recommendation score
- Future pull requests will add onto this by introducing comparisons of recent GitHub repositories (comparing language and topics (aka tags) for example) 
- Utilized and initialized the connections graph upon starting the server
- Fewer fetch calls to database because of graph

## Milestones & User Stories
- User can see recommended posts and profiles

## Test Plan
_No dramatic changes on frontend, but video of key features on Loom:_
https://www.loom.com/share/40a20b3549ac4be69eb86f39f1c29b2d?sid=8c519c45-966c-420c-bc42-1e10b6cb415e

_Edge Cases Tested:_
- Adjacent connections already being connected to the recommendee user

### CODE REVIEW

File: backend/endpoints/connections.js [hunk 1]
```diff
@@ -1,22 +1,72 @@
 // Endpoints and graph for connections.
 
-module.exports = function (app) {
+const { connected } = require("process");
```
Comment (TsukkiGia) [inline backend/endpoints/connections.js:3]: Did you import this by mistake?

Comment (dezok9) [reply backend/endpoints/connections.js:3]: Was going to use, but didn't end up doing so. Thanks for catching.
